### PR TITLE
Add stockItem implementation

### PR DESCRIPTION
**Pull request by RUKAIYIN**

### CODE REVIEW

File: stock/src/main/java/nl/tudelft/wdm/group1/stock/StockItem.java [hunk 1]
```diff
@@ -1,22 +1,48 @@
...
     }
 
     public UUID getId() {
         return id;
     }
 
+    public int getStock() { return this.stock; }
+
+    public void addStock(final int amount) {
+        if (amount > 0) this.stock += amount; // TODO: maybe create a new exception for this?
```
Comment (casperboone) [inline stock/src/main/java/nl/tudelft/wdm/group1/stock/StockItem.java]: Why do we need this if statement here?

File: stock/src/main/java/nl/tudelft/wdm/group1/stock/StockItem.java [hunk 2]
```diff
@@ -1,22 +1,48 @@
...
+    }
+
+    public void subtractStock(final int amount) throws InsufficientStockException{
+        if (amount > this.stock) {
+            throw new InsufficientStockException("Insufficient stock");
+        }
+
+        if (amount > 0) {
+            this.stock -= amount;
+        }
```
Comment (casperboone) [inline stock/src/main/java/nl/tudelft/wdm/group1/stock/StockItem.java]: else?

File: stock/src/main/java/nl/tudelft/wdm/group1/stock/web/StockController.java [hunk 3]
```diff
@@ -39,4 +36,29 @@ public StockItem addStockItem() {
...
     }
+
+    @PostMapping("/{id}/subtract/{amount}")
+    public StockItem substractStockItemAmount(
+            @PathVariable(value = "id") UUID id,
+            @PathVariable(value = "amount") int amount
+    ) throws ResourceNotFoundException, InsufficientStockException {
+        StockItem item = stockItemRepository.find(id);
+        item.subtractStock(amount);
+        producer.send(item);
```
Comment (casperboone) [inline stock/src/main/java/nl/tudelft/wdm/group1/stock/web/StockController.java]: Subtraction/addition need different events, see https://github.com/casperboone/wdm-kafka-microservices/pull/2

File: stock/src/main/java/nl/tudelft/wdm/group1/stock/web/StockController.java [hunk 4]
```diff
@@ -39,4 +36,29 @@ public StockItem addStockItem() {
...
+    }
+
+    @PostMapping("{id}/add/{amount}")
+    public StockItem addStockItemAmount(
+            @PathVariable(value = "id") UUID id,
+            @PathVariable(value = "amount") int amount
+    ) throws ResourceNotFoundException {
+        StockItem item = stockItemRepository.find(id);
+        item.addStock(amount);
+        producer.send(item);
```
Comment (casperboone) [inline stock/src/main/java/nl/tudelft/wdm/group1/stock/web/StockController.java]: Subtraction/addition need different events, see https://github.com/casperboone/wdm-kafka-microservices/pull/2

File: stock/src/main/java/nl/tudelft/wdm/group1/stock/StockItem.java (hunk 1 above)
Comment (RUKAIYIN) [reply stock/src/main/java/nl/tudelft/wdm/group1/stock/StockItem.java]: I want to make sure the amount is positive

File: stock/src/main/java/nl/tudelft/wdm/group1/stock/StockItem.java (hunk 2 above)
Comment (RUKAIYIN) [reply stock/src/main/java/nl/tudelft/wdm/group1/stock/StockItem.java]: It does nothing if you subtract a negative number. I omitted the else

File: stock/src/main/java/nl/tudelft/wdm/group1/stock/web/StockController.java (hunk 3 above)
Comment (RUKAIYIN) [reply stock/src/main/java/nl/tudelft/wdm/group1/stock/web/StockController.java]: Got it.

File: stock/src/main/java/nl/tudelft/wdm/group1/stock/StockItem.java (hunk 1 above)
Comment (casperboone) [reply stock/src/main/java/nl/tudelft/wdm/group1/stock/StockItem.java]: Ahh, I misread this as `if (stock > 0)`

In that case I would throw an exception in the else case. See https://github.com/casperboone/wdm-kafka-microservices/blob/master/users/src/main/java/nl/tudelft/wdm/group1/users/User.java for an example (credit subtraction/addition is very similar)

File: stock/src/main/java/nl/tudelft/wdm/group1/stock/StockItem.java (hunk 2 above)
Comment (casperboone) [reply stock/src/main/java/nl/tudelft/wdm/group1/stock/StockItem.java]: See previous comment about throwing an exception

File: stock/src/main/java/nl/tudelft/wdm/group1/stock/StockItem.java [hunk 5]
```diff
@@ -1,22 +1,48 @@
 package nl.tudelft.wdm.group1.stock;
 
+import jdk.nashorn.internal.objects.annotations.Constructor;
```
Comment (casperboone) [inline stock/src/main/java/nl/tudelft/wdm/group1/stock/StockItem.java]: Seems to be unused

Comment (RUKAIYIN) [reply stock/src/main/java/nl/tudelft/wdm/group1/stock/StockItem.java]: True. Not sure why it's there.

File: stock/src/main/java/nl/tudelft/wdm/group1/stock/StockItem.java (hunk 1 above)
Comment (RUKAIYIN) [reply stock/src/main/java/nl/tudelft/wdm/group1/stock/StockItem.java]: alright then

**Pull request merged**